### PR TITLE
Support function from v2

### PR DESCRIPTION
**Pull request by hoangdat**
Cherry pick function from v2:
- create workgroup
- enhance upload outside
- add member
- bugs: #263, #310, #311, #312

### CODE REVIEW

**Comment by hoangdat** [comment]
@dab246 @huynguyennovem please review it.

File: lib/presentation/util/extensions/suggest_name_type_extension.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:domain/domain.dart';
+import 'package:flutter/material.dart';
+import 'package:linshare_flutter_app/presentation/localizations/app_localizations.dart';
+
+const INDEX_1 = 1;
+const UP_TO_NEXT_INDEX = 1;
```
Comment (huynguyennovem) [inline lib/presentation/util/extensions/suggest_name_type_extension.dart:37]: this const is redundant. should remove it?

File: lib/presentation/util/extensions/suggest_name_type_extension.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  String getName(BuildContext context) {
+    switch(this) {
+      case SuggestNameType.WORKGROUP:
+        return AppLocalizations.of(context).new_workgroup;
+      default:
+        return '';
+    }
+  }
+
+  String patternNameWithIndex(String originalName, int index) {
```
Comment (huynguyennovem) [inline lib/presentation/util/extensions/suggest_name_type_extension.dart:50]: This method are not dependent on SudependentggestNameType. They should be string extensions.

File: lib/presentation/util/extensions/suggest_name_type_extension.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    return '${originalName} (${index})';
+  }
+
+  String suggestNameWithIndex(String originalSuggestName, List<String> listName) {
```
Comment (huynguyennovem) [inline lib/presentation/util/extensions/suggest_name_type_extension.dart:54]: same above

**Review by dab246 — APPROVED** [review]
LGTM

**Review by huynguyennovem — APPROVED** [review]
/lgtm

**Pull request merged**